### PR TITLE
Add rockylinux support

### PR DESCRIPTION
**Pull request by Dean-Coakley**
### Proposed changes

* Add support for rocky linux packages
* Extend integration test to support rockylinux base image
* Extend api integration test to run on centos based OSes

### TODO
- [x] Dockerfile/Makefile structure
- [x] Update CI to run integration tests on many base images

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [`CONTRIBUTING`](https://github.com/nginx/agent/blob/main/docs/CONTRIBUTING.md) document
- [x] If applicable, I have added tests that prove my fix is effective or that my feature works
- [x] If applicable, I have checked that any relevant tests pass after adding my changes
- [x] I have updated any relevant documentation ([`README.md`](https://github.com/nginx/agent/blob/main/README.md))

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -64,18 +64,37 @@ jobs:
       - name: Run Component Tests
         run: make component-test
 
-  integration-test:
+  integration-test-deb:
     name: Integration Tests
```
Comment (dhurley) [inline .github/workflows/ci.yml:68]: Can you update the name to `Debian Integration Tests`?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -64,18 +64,37 @@ jobs:
...
           go-version-file: 'go.mod'
       - name: Run Integration Tests
         run: |
-            go install github.com/goreleaser/nfpm/v2/cmd/nfpm@${{ env.NFPM_VERSION }} 
-            make integration-test
+          go install github.com/goreleaser/nfpm/v2/cmd/nfpm@${{ env.NFPM_VERSION }}
+          BASE_IMAGE="docker.io/${{ matrix.base-image }}" DOCKER_COMPOSE_FILE="docker-compose.yml" make integration-test
+
+  integration-test-rpm:
+    name: Integration Tests
```
Comment (dhurley) [inline .github/workflows/ci.yml:84]: Can you update the name to `Red Hat Integration Tests`?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -64,18 +64,37 @@ jobs:
...
       - uses: actions/checkout@v3
       - uses: actions/setup-go@v3
         with:
           go-version-file: 'go.mod'
       - name: Run Integration Tests
         run: |
-            go install github.com/goreleaser/nfpm/v2/cmd/nfpm@${{ env.NFPM_VERSION }} 
-            make integration-test
+          go install github.com/goreleaser/nfpm/v2/cmd/nfpm@${{ env.NFPM_VERSION }}
+          BASE_IMAGE="docker.io/${{ matrix.base-image }}" DOCKER_COMPOSE_FILE="docker-compose.yml" make integration-test
```
Comment (dhurley) [inline .github/workflows/ci.yml:81]: Can you rename the `docker-compose.yml` to `docker-compose-deb.yml` to be consistent?

File: Makefile [hunk 4]
```diff
@@ -179,9 +182,9 @@ test-component-run: ## Run component tests
 performance-test: ## Run performance tests
 	$(CONTAINER_CLITOOL) run -v ${PWD}:/home/nginx/$(CONTAINER_VOLUME_FLAGS) --rm nginx-agent-benchmark:1.0.0
 
-integration-test: local-deb-package
-	PACKAGE_NAME=${PACKAGE_NAME} BASE_IMAGE=${BASE_IMAGE} go test ./test/integration/install
-	PACKAGE_NAME=${PACKAGE_NAME} BASE_IMAGE=${BASE_IMAGE} go test ./test/integration/api
+integration-test: local-deb-package local-rpm-package
+	PACKAGE_NAME=${PACKAGE_NAME} BASE_IMAGE=$(BASE_IMAGE) DOCKER_COMPOSE_FILE=$(TEST_DOCKER_COMPOSE_FILE) go test -v ./test/integration/install
```
Comment (dhurley) [inline Makefile:186]: Do you need the `-v` option here?

**Review by aphralG — APPROVED** [review]

File: scripts/docker/nginx-oss/centos/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,39 @@
+ARG BASE_IMAGE
```
[the remark this answers is not in the document]
Comment (Dean-Coakley) [reply scripts/docker/nginx-oss/centos/Dockerfile:1]: Not sure. In theory it should be able to use any base image that uses the `yum` package manager.

I'll change it to `rockylinux` for now. If it becomes generic in the future we can rename at that point

File: Makefile.packaging [hunk 6]
```diff
@@ -36,27 +38,45 @@ package: gpg-key $(PACKAGES_DIR) $(GITHUB_PACKAGES_DIR) ## Create final packages
...
+	@for arch in $(ROCKY_ARCHS); do \
+		GOWORK=off CGO_ENABLED=0 GOARCH=$${arch} GOOS=linux go build -ldflags=${LDFLAGS} -o ./build/nginx-agent; \
+		for distro in $(ROCKY_VERSIONS); do \
+			rpm_distro=`echo $$distro | cut -d- -f 1`; \
+			rpm_major=`echo $$distro | cut -d- -f 2`; \
+			rpm_codename='na'; \
+			if [ "$$rpm_distro" = "rocky" ]; then rpm_codename="rocky$$rpm_major"; fi; \
+			if [ "$$rpm_codename" != "na" ]; then \
+				VERSION=$(shell echo ${VERSION} | tr -d 'v') ARCH=$${arch} nfpm pkg --config .nfpm.yaml --packager rpm --target $(PACKAGES_DIR)/rpm/${PACKAGE_PREFIX}-$(shell echo ${VERSION} | tr -d 'v').$${rpm_codename}.ngx.$${arch}.rpm; \
+				cp $(PACKAGES_DIR)/rpm/${PACKAGE_PREFIX}-$(shell echo ${VERSION} | tr -d 'v').$${rpm_codename}.ngx.$${arch}.rpm ${GITHUB_PACKAGES_DIR}/${PACKAGE_PREFIX}-$(shell echo ${VERSION} | tr -d 'v').$${rpm_codename}.ngx.$${arch}.rpm; \
```
[the remark this answers is not in the document]
Comment (Dean-Coakley) [reply Makefile.packaging:80]: Seems it is wrong for deb builds too? https://github.com/nginx/agent/blob/cce39148ba5f027055d8eaac138221af2dc09318/Makefile.packaging#L11

Will I fix as part of this PR?

**Review by dhurley — APPROVED** [review]

**Review by craigell — APPROVED** [review]

**Pull request merged**